### PR TITLE
fix(input-date-picker): Set value, start, and end properties via input or date-picker. #2955

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #2955

## Summary

fix(input-date-picker): Set value, start, and end properties via input or date-picker. #2955

### CODE REVIEW

File: src/components/calcite-input-date-picker/calcite-input-date-picker.tsx [hunk 1]
```diff
@@ -501,49 +508,54 @@ export class CalciteInputDatePicker {
     this.localeData = await getLocaleData(locale);
   }
 
-  /**
-   * Update date instance of start if valid
-   */
-  private setStartAsDate(startDate: Date): void {
-    this.startAsDate = startDate;
-  }
+  private clearCurrentValue() {
```
Comment (jcfranco) [inline src/components/calcite-input-date-picker/calcite-input-date-picker.tsx:511]: Can you specify the return type?

File: src/components/calcite-input-date-picker/calcite-input-date-picker.tsx [hunk 2]
```diff
@@ -194,11 +201,11 @@ export class CalciteInputDatePicker {
     }
 
     if (this.start) {
-      this.setStartAsDate(dateFromISO(this.start));
+      this.startAsDate = dateFromISO(this.start);
```
Comment (jcfranco) [inline src/components/calcite-input-date-picker/calcite-input-date-picker.tsx:204]: Noice cleanup! ✨🧹 ✨

File: src/components/calcite-input-date-picker/calcite-input-date-picker.e2e.ts [hunk 3]
```diff
@@ -85,6 +88,11 @@ describe("calcite-input-date-picker", () => {
     await input.press("0");
     await page.waitForChanges();
     expect(changedEvent).toHaveReceivedEventTimes(4);
+    const element = await page.find("calcite-input-date-picker");
+    element.setProperty("end", "2020-03-05");
+    await page.waitForChanges();
+    expect(await element.getProperty("start")).toBe("2020-03-07");
+    expect(await element.getProperty("startAsDate")).toBeDefined();
```
Comment (jcfranco) [inline src/components/calcite-input-date-picker/calcite-input-date-picker.e2e.ts:95]: Can you assert the type instead of a non-undefined value?

**Review by jcfranco — APPROVED** [review]
💥

File: src/components/calcite-input-date-picker/calcite-input-date-picker.e2e.ts (hunk 3 above)
Comment (driskull) [reply src/components/calcite-input-date-picker/calcite-input-date-picker.e2e.ts:95]: I tried but it looks like the e2e tests don't actually return a date.

**Pull request merged**